folium/plugins/polyline_text_path.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""
+PolyLine Text Path
+-----------
+
+Shows a text along a PolyLine.
+
+"""
+from jinja2 import Template
+
+from branca.element import JavascriptLink, Figure
+from folium.features import MacroElement
+
+
+class PolyLineTextPath(MacroElement):
+    """Shows a text along a PolyLine."""
+    def __init__(self, polyline, text, repeat=False, center=False, below=False, offset=0, orientation=0, attributes={}):
+        """Shows a text along a PolyLine.
+
+        Parameters
+        ----------
+            polyline: folium.features.PolyLine object
+                The folium.features.PolyLine object to attach the text to.
+
+            text: string
+                The string to be attached to the polyline.
+
+            repeat: bool, default False
+                Specifies if the text should be repeated along the polyline.
+
+            center: bool, default False
+                Centers the text according to the polyline's bounding box
+
+            below: bool, default False
+                Show text below the path
+
+            offset: int, default 0
+                Set an offset to position text relative to the polyline.
+
+            orientation: int, default 0
+                Rotate text to a specified angle.
+
+            attributes: dictionary
+                Object containing the attributes applied to the text tag.
+                Check valid attributes here: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/text#Attributes
+                Example: {'fill': '#007DEF', 'font-weight': 'bold', 'font-size': '24'}
+
```

Comment:
Please add a note and the URL for the plugin page. (I guess https://github.com/makinacorpus/Leaflet.TextPath ?)